pages/student/home.vue

```diff
@@ -31,97 +54,129 @@
 
 <script>
 import NavigationBar from "~/components/NavigationBar.vue";
-import HorizontalScroll from "~/components/HorizontalScroll.vue";
+// import HorizontalScroll from "~/components/HorizontalScroll.vue";
 import CourseCard from "~/components/CourseCard";
+import courses from '~/services/http/courses';
 
 export default {
-  data: () => ({
-    categoryCourses:[{
-      name: "Cursos",
-      courses:[
-        {
-          id:1,
-          name: "Curso 1",
-          teacher: "Gustavo Raña",
-          image: "http://i.imgur.com/SrPdUD4.png"
-        },
-        {
-          id:2,
-          name: "Curso 2",
-          teacher: "Mateus Vinicius",
-          image: "https://s31450.pcdn.co/wp-content/uploads/2015/10/iStock_computer-art.151110.jpg"
-        },
-        {
-          id:3,
-          name: "Curso 3",
-          teacher: "Patricia Camarões",
-          image: "http://i.imgur.com/SrPdUD4.png"
-        },
-      ]
-    },
-    {
-      name: "Cursos top",
-      courses:[
-        {
-          id:4,
-          name: "Curso 1",
-          teacher: "Gustavo Raña",
-          image: "https://ugc.futurelearn.com/uploads/images/41/22/regular_4122fa44-6ac3-4c39-ba9b-29a7160e763b.jpg"
-        },
-        {
-          id:5,
-          name: "Curso 2",
-          teacher: "Mateus Vinicius",
-          image: "https://s31450.pcdn.co/wp-content/uploads/2015/10/iStock_computer-art.151110.jpg"
-        },
-        {
-          id:6,
-          name: "Curso 3",
-          teacher: "Patricia Camarões",
-          image: "http://i.imgur.com/SrPdUD4.png"
-        },
-      ]
-    },
-    {
-      name: "Cursos mais vistos",
-      courses:[
-        {
-          id:7,
-          name: "Curso 1",
-          teacher: "Gustavo Raña",
-          image: "https://s31450.pcdn.co/wp-content/uploads/2015/10/iStock_computer-art.151110.jpg"
-        },
-        {
-          id:8,
-          name: "Curso 2",
-          teacher: "Mateus Vinicius",
-          image: "https://ugc.futurelearn.com/uploads/images/41/22/regular_4122fa44-6ac3-4c39-ba9b-29a7160e763b.jpg"
-        },
-        {
-          id:9,
-          name: "Curso 3",
-          teacher: "Patricia Camarões",
-          image: "http://i.imgur.com/SrPdUD4.png"
-        },
-      ]
-    },
-    ]
-  }),
   components: {
     NavigationBar,
-    HorizontalScroll,
     CourseCard
+  },
+  data: () => ({
+    user: {
+      name: ''
+    }
+  }),
+  computed: {
+    courses() {
+      return this.$store.state.courses.list
+    }
+  },
+  methods: {
+    getUser() {
+      let storedUser = JSON.parse(localStorage.getItem("user"))
+      if (storedUser) {
+        this.user = storedUser;
+        this.user.name = this.user.name.split(' ')[0]
+      }
+    }
+  },
+  mounted() {
+    this.getUser();
+  },
+  asyncData({ store, data, params, $axios }) {
+    const coursesPromise = courses.getAll().then(response =>
```

Comment:
```suggestion
    return courses.getAll().then(response =>
```
Aqui já tem uma Promise, só necessário retornar.
Quando usamos Promise.resolve normalmente é para retornar um valor como uma promise, mas não algo que já é uma Promise.

Reply:
É verdade, eu ia fazer mais de uma chamada ao back, e usar o Promise.all(), mas como acabou mantendo apenas uma eu esqueci de alterar o bloco.